### PR TITLE
pipewire: 0.3.25 -> 0.3.26

### PR DESCRIPTION
**Pull request by talyz**
###### Motivation for this change
- Update `pipewire` to the latest version. See the [changelog](https://gitlab.freedesktop.org/pipewire/pipewire/-/raw/f0fc4f7fc644c29762fa34e57e60d4b5cbcefb47/NEWS) for details.
- Fix the tests
- Add an update script

###### Things done
- Tested manually

- [x] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [x] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

**Comment by legendofmiracles** [comment]
Duplicate of https://github.com/NixOS/nixpkgs/pull/120748

**Comment by hmenke** [comment]
I suggested to include https://github.com/NixOS/nixpkgs/pull/120833/commits/d393845335ed427f38144ce41aea4153120b9de8 in #120748.

**Comment by legendofmiracles** [comment]
Since they're on the other PR, can this be closed?

**Comment by talyz** [comment]
I'd prefer to just rebase this on #120748 and remove the package update, since this now also includes an update script.

**Comment by balsoft** [comment]
I would be OK with any way to get 0.3.26 in nixpkgs, I just really wanted hardware volume to work :)

If you want to, I can close my PR and just merge yours.

**Comment by jansol** [comment]
Either PR seems fine (I'm still confused about the hash syntax policy but whatever). The update script is a step in the right direction so that would be nice to have. However a more common case than new config files getting added seems to be that the available keys change, so the tests should also compare found files and warn about differences.

File: pkgs/development/libraries/pipewire/update.sh [hunk 1]
```diff
@@ -0,0 +1,24 @@
+#!/usr/bin/env nix-shell
+#!nix-shell -p nix-update -i bash
+# shellcheck shell=bash
+
+set -o errexit -o pipefail -o nounset -o errtrace
+shopt -s inherit_errexit
+shopt -s nullglob
+IFS=$'\n'
+
+NIXPKGS_ROOT="$(dirname "$(readlink -f "${BASH_SOURCE[0]}")")/../../../.."
```
Comment (hmenke) [inline pkgs/development/libraries/pipewire/update.sh:10]: Just a suggestion
```suggestion
NIXPKGS_ROOT="$(git rev-parse --show-toplevel)"
```
This assumes that this is run in a Git checkout of nixpkgs but that is most likely the case.

Comment (talyz) [reply pkgs/development/libraries/pipewire/update.sh]: Ah, that's true. Thanks!

**Comment by talyz** [comment]
@balsoft Sure, I'm fine either way :)

@jansol I'm not really sure what you mean by that - since the keys are set by the user directly, there isn't much we can do if the schema changes, right? Am I missing something?

**Comment by collares** [comment]
Clarifying the concern: Suppose upstream edits a config file (not adding a new file, just editing an existing one). Ideally, tests would fail if someone updates pipewire manually (without the update script) and forgets to copy the new files to the NixOS module.

**Comment by jansol** [comment]
Upstream has renamed config keys that are present in the default config files and will likely do that again in the future. This would make an old unmodified default config no longer work the intended way so it's something I'd like to detect. Users' customized configs are their own problem and the tests aren't concerned with those.

**Comment by talyz** [comment]
@jansol Well, the `installedTests` test should show this, right? It enables a "fully featured" pipewire config and runs the test suite - if the default config is severely broken, it should fail. However, the update script should make sure this doesn't happen as often or at all going forward.

**Comment by talyz** [comment]
Anything more to do here or can we merge?

**Comment by jansol** [comment]
Works fine on my end, should be good to go.

**Pull request merged**